src/main/java/logisticspipes/renderer/LogisticsRenderPipe.java

```diff
@@ -136,12 +140,17 @@ private void renderSolids(CoreUnroutedPipe pipe, double x, double y, double z, f
                 break;
             }
 
-            LPPosition pos = new LPPosition(0.5D, 0.5D, 0.5D);
+            if (item.getPosition() > 1 || item.getPosition() < 0 || item.getItemIdentifierStack() == null) {
+                continue;
+            }
 
-            if (item.getPosition() > 1 || item.getPosition() < 0) {
+            if (item.getContainer().xCoord != pipe.container.xCoord
+                    || item.getContainer().yCoord != pipe.container.yCoord
+                    || item.getContainer().zCoord != pipe.container.zCoord) {
                 continue;
             }
 
+            pos.reset(0.5D, 0.5D, 0.5D);
             float fPos = item.getPosition() + item.getSpeed() * partialTickTime;
 
             if (fPos < 0.5) {
```

Comment:
The only side effect in this if/else is the `pos.moveForward(item....`) and the result has no bearing on the `continue` checks that come after. That is why it's safe to move the `continue` checks above the `pos` calculation.